app/controllers/teachers_controller.rb

```diff
@@ -51,6 +51,12 @@ def edit
     @readonly = !is_admin?
   end
 
+  def show
```

Comment:
I could be wrong, but I think you need to add :show to the require_admin precondition on top of this file.